fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -753,13 +757,13 @@ public FHIRRestOperationResponse doDelete(String type, String id, String searchQ
     }
 
     @Override
-    public Resource doRead(String type, String id, boolean throwExcOnNull, boolean includeDeleted,
+    public SingleResourceResult<? extends Resource> doRead(String type, String id, boolean throwExcOnNull, boolean includeDeleted,
             Resource contextResource) throws Exception {
         return doRead(type, id, throwExcOnNull, includeDeleted, contextResource, null);
     }
 
     @Override
-    public Resource doRead(String type, String id, boolean throwExcOnNull, boolean includeDeleted,
+    public SingleResourceResult<? extends Resource> doRead(String type, String id, boolean throwExcOnNull, boolean includeDeleted,
```

Comment:
Do we still need to pass `includeDeleted` now that we have the `isDeleted` marker in the response?

Reply:
It would be possible to refactor the entire handling of deleted status, but this is a much bigger task. Currently the persistence layer will throw an exception to signal the resource has been deleted. Handling this without using an exception is non-trivial.